cpg_pipes/providers/cpg/inputs.py

```diff
@@ -0,0 +1,190 @@
+"""
+InputProvider implementation that pulls data from the sample-metadata database.
+"""
+
+import logging
+
+from .smdb import SMDB, SmSequence
+from ..inputs import InputProvider, InputProviderError
+from ... import Path
+from ...targets import Cohort, Sex, PedigreeInfo, Dataset
+from ...types import SequencingType
+
+logger = logging.getLogger(__file__)
+
+
+class SmdbInputProvider(InputProvider):
+    """
+    InputProvider implementation that pulls data from the sample-metadata database.
+    """
+
+    def __init__(self, db: SMDB, check_files: bool = False):
+        super().__init__(check_files=check_files)
+        self.db = db
+
+    def populate_cohort(
+        self,
+        cohort: Cohort,
+        dataset_names: list[str] | None = None,
```

Comment:
Here and below, what's preferable in terms of Python style? `type | None` or `Optional[type]`?

Reply:
`type | None` is the new syntax, introduced in python 3.10 https://peps.python.org/pep-0604, and should become preferable as far as I understand. The goal is to allow to avoid importing the `typing` module for most cases (that's why python 3.10 also now allows using `dict`, `list`, `tuple` as type hints instead of `Typing.List`, `Typing.Dict` etc.)